igcviewer.js

```diff
@@ -59,12 +59,34 @@
         }
 
         // Show the task declaration if it is present.
-        if (igcFile.task.coordinates.length > 0) {
+        //First two coordinates are always takeoff and start. Last two always finish and landing
+        //So there must be at least 5 for a vailid declaration
+        if (igcFile.task.coordinates.length > 4) {
             $('#task').show();
-            var taskList = $('#task ol').first().html('');
+            var taskList = $('#task ul').first().html('');
             var j;
+            //Now add TP numbers.  Change to unordered list
             for (j = 0; j < igcFile.task.names.length; j++) {
-                taskList.append($('<li></li>').text(igcFile.task.names[j]));
+                switch(j)  {
+                    case 0:
+                          if (igcFile.task.coordinates[0][0] != 0.0 && igcFile.task.coordinates[0][1] != 0.0) {
```

Comment:
JavaScript has a subtle design problem with the `==` and `!=` operators. If the two operands are of different types the the interpreter will try to coerce them to be of the same type, sometimes with odd results. There are some weird situations where `a == b` is true but `b == a` is false!

It is therefore generally safer always to use `===` and `!==`, which do not coerce the types of the operands and always do what you would expect.

For details see [JavaScript - The Bad Parts.](http://archive.oreilly.com/pub/a/javascript/excerpts/javascript-good-parts/bad-parts.html)